pkg/k8shandler/secret_test.go

```diff
@@ -0,0 +1,67 @@
+-----END PRIVATE KEY-----`
+
+    initSecret := NewSecret(
+        "test-secret",
+        clusterLoggingRequest.cluster.Namespace,
+        map[string][]byte{
+            "testca":  []byte(caCrtStr),
+            "testkey": []byte(caKeyStr),
```

Comment:
I think 4 test cases:
- verify that trying to update an existing secret that is identical is a no-op
- verify that trying to update an existing secret with a key that has a different value will update the secret
- verify that updating a secret with a new key will update the secret
- verify that updating a secret by removing a key will update the secret